modules/nf-scil/betcrop/fslbetcrop/main.nf

```diff
@@ -0,0 +1,75 @@
+
+process BETCROP_FSLBETCROP {
+    tag "$meta.id"
+    label 'process_single'
+
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://scil.usherbrooke.ca/containers/scilus_1.5.0.sif':
+        'scilus/scilus:1.5.0' }"
+
+    input:
+        tuple val(meta), path(dwi), path(bval), path(bvec)
+
+    output:
+        tuple val(meta), path("*dwi_bet_cropped.nii.gz")            , emit: dwi
+        tuple val(meta), path("*dwi_bet_cropped_mask.nii.gz")       , emit: mask
+        tuple val(meta), path("*dwi_boundingBox.pkl")               , emit: bbox
+        path "versions.yml"                                         , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def prefix = task.ext.prefix ?: "${meta.id}"
+
+    def b0_thr = task.ext.b0_thr ? "--b0_thr " + task.ext.b0_thr : ""
+    def bet_dwi_f = task.ext.bet_dwi_f ? "-f " + task.ext.bet_dwi_f : ""
+
+    """
+    export ITK_GLOBAL_DEFAULT_NUMBER_OF_THREADS=1
+    export OMP_NUM_THREADS=1
+    export OPENBLAS_NUM_THREADS=1
+
+    scil_extract_b0.py $dwi $bval $bvec ${prefix}__b0.nii.gz --mean \
+        $b0_thr --force_b0_threshold
+    bet ${prefix}__b0.nii.gz ${prefix}__b0_bet.nii.gz -m -R $bet_dwi_f
+    scil_image_math.py convert ${prefix}__b0_bet_mask.nii.gz ${prefix}__b0_bet_mask.nii.gz --data_type uint8 -f
+    mrcalc $dwi ${prefix}__b0_bet_mask.nii.gz -mult ${prefix}__dwi_bet.nii.gz -quiet -nthreads 1
+
+    scil_crop_volume.py $dwi ${prefix}__dwi_bet_cropped.nii.gz -f \
+        --output_bbox ${prefix}__dwi_boundingBox.pkl -f
+    scil_crop_volume.py ${prefix}__b0_bet_mask.nii.gz ${prefix}__dwi_bet_cropped_mask.nii.gz -f\
+        --input_bbox ${prefix}__dwi_boundingBox.pkl -f
+    scil_image_math.py convert ${prefix}__dwi_bet_cropped_mask.nii.gz ${prefix}__dwi_bet_cropped_mask.nii.gz \
+        --data_type uint8 -f
+
+    cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
+        : \$(scil_get_version.py 2>&1)
+        : \$(mrcalc -version 2>&1)
+        : \$(flirt -version 2>&1)
+    END_VERSIONS
```

Comment:
Last modification I just remembered of ! I got this part wrong for the version (and it is probably why the version test is not passing). Add the dependency name before the `:` (e.g. : `scilpy: \$(scil_get_version ... `).

Also, I just realized that `scil_get_version.py` does not exists in scilus:1.5.0. You can change the containers versions you use to 1.6.0, they have the script, or simply replace `\(scil_get_version.py 2>&1)` by `1.5.0` for now.

Reply:
Okay, I think I figured out what was crashing. I tried your suggestion (bumping scilus:1.5.0 to scilus:1.6.0 and adding `scilpy: \$(scil_get_version ...`) and actually got further in the test execution. The error message gave me more details this time :
```
        for level in list(self.possible_simple_keys):
            key = self.possible_simple_keys[level]
            if key.line != self.line  \
                    or self.index-key.index > 1024:
                if key.required:
>                   raise ScannerError("while scanning a simple key", key.mark,
                            "could not find expected ':'", self.get_mark())
E                   yaml.scanner.ScannerError: while scanning a simple key
E                     in "<unicode string>", line 4, column 1:
E                       64 bit release version with nogu ...
E                       ^
E                   could not find expected ':'
E                     in "<unicode string>", line 5, column 1:
E                       Author(s): J-Donald Tournier (jd ...
E                       ^

../../../.local/pipx/venvs/nf-core/lib/python3.11/site-packages/yaml/scanner.py:291: ScannerError
```

It seems the test_version_yml.py is expecting the version to be numeric value (or something like a classic versioning number, ex: 1.5.0 for scilpy). Testing with manually placing version number like this :
```
    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        scilpy: 1.6.0
        mrtrix: 3.0.4
        fsl: 6.0
    END_VERSIONS
    """
```
Passed without any errors ! So I replaced the manual values with this :
```
    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        scilpy: 1.6.0
        mrtrix: \$(mrcalc -version 2>&1 | sed -n 's/== mrcalc \\([0-9.]\\+\\).*/\\1/p')
        fsl: \$(flirt -version 2>&1 | sed -n 's/FLIRT version \\([0-9.]\\+\\)/\\1/p')
    END_VERSIONS
```
Passed with flying colors! For some reason, I couldn't use the same pattern recognition for scilpy since `docker run scilus/scilus:1.6.0 scil_get_version.py` is returning git related errors, but I think it is more scilpy related than nf-scil. Feel free to test it out!

Reply:
Very nice, thank you for investigating ! I will take a note to fix scil_get_version. I think we can set this as successful, I will use your solution to fix the other PR.